web-registry/src/pages/MyEcocredits.tsx

```diff
@@ -63,34 +130,117 @@ export const MyEcocredits: React.FC = () => {
+              // eslint-disable-next-line no-console
+              onClick: () => console.log(`TODO retire credit ${i}`),
             },
-          ]}
-        />
-      )}
-      renderBasketActionButtons={(i: number) => (
-        <TableActionButtons
-          buttons={[
-            {
-              icon: <TakeFromBasket />,
-              label: 'Take from basket',
-              onClick: () => `TODO take from basket ${i}`,
-            },
-            // This will be handled from osmosis
-            // so hiding these for now
-            // {
-            //   icon: <WithdrawIBC />,
-            //   label: 'Withdraw (IBC)',
-            //   onClick: () => `TODO withdraw ${i}`,
-            // },
-            // {
-            //   icon: <DepositIBC />,
-            //   label: 'Deposit (IBC)',
-            //   onClick: () => `TODO deposit ${i}`,
-            // },
-          ]}
+          ];
+
+          // Only add ability to put credits into basket
+          // if there's at least one basket that accepts those credits
+          if (creditBaskets[i] && creditBaskets[i].length > 0) {
+            buttons.splice(1, 0, {
+              // buttons.splice(2, 0, { TODO: Replace once we had 'Sell'
+              icon: <PutInBasket />,
+              label: 'Put in basket',
+              onClick: () => setBasketPutOpen(i),
+            });
+          }
+          return <TableActionButtons buttons={buttons} />;
+        }}
+        renderBasketActionButtons={(i: number) => (
+          <TableActionButtons
+            buttons={[
+              {
+                icon: <TakeFromBasket />,
+                label: 'Take from basket',
+                // eslint-disable-next-line no-console
+                onClick: () => console.log(`TODO take from basket ${i}`),
+              },
+              // This will be handled from osmosis
+              // so hiding these for now
+              // {
+              //   icon: <WithdrawIBC />,
+              //   label: 'Withdraw (IBC)',
+              //   onClick: () => `TODO withdraw ${i}`,
+              // },
+              // {
+              //   icon: <DepositIBC />,
+              //   label: 'Deposit (IBC)',
+              //   onClick: () => `TODO deposit ${i}`,
+              // },
+            ]}
+          />
+        )}
+      />
+      {basketPutOpen > -1 && (
+        <BasketPutModal
+          basketOptions={
+            creditBaskets[basketPutOpen]
+              .map(b => ({
+                label: b?.basket?.name,
+                value: b?.basket?.basketDenom,
+              }))
+              .filter(v => v.label && v.value) as Option[]
+          }
+          availableTradableAmount={Number(
+            credits[basketPutOpen].tradable_amount,
+          )}
+          batchDenom={credits[basketPutOpen].batch_denom}
+          open={basketPutOpen > -1}
+          onClose={() => setBasketPutOpen(-1)}
+          onSubmit={async (values: BasketPutFormValues) => {
```

Comment:
Just a nit. I prefer to have logic above the `render` or `return`. In a method like `handlePutSubmit`. It's a separation-of-concerns pattern I've always followed. This could be a style discussion for later, however.